Libraries/process/Tests/test_DefaultLauncher.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <gtest/gtest.h>
+#include <process/DefaultLauncher.h>
+#include <process/LaunchResult.h>
+#include <process/MemoryContext.h>
+#include <libutil/DefaultFilesystem.h>
+
+#include <unistd.h>
+
+using process::DefaultLauncher;
+using process::MemoryContext;
+using libutil::DefaultFilesystem;
+
+TEST(DefaultLauncher, CapturesStdout)
+{
+    DefaultLauncher launcher;
+    auto filesystem = DefaultFilesystem::GetDefaultUNSAFE();
```

Comment:
Yeah, I had a bit of trouble here. I figured that I shouldn't use a method suffixed with `UNSAFE`, but I couldn't figure out how to use `MemoryFilesystem`. It seems like I can instantiate a `MemoryFilesystem` with files and directories, but not executables that would print to stdout. Other tests like `xcexecution/Tests/test_SimpleExecutor.cpp` use `process::MemoryLauncher` handlers, but that doesn't fit my use case, since I'd like to test `process::DefaultLauncher` here.

Another interesting approach taken in `test_SimpleExecutor.cpp` is using `builtin::Registry::Create` to instantiate executables, but that might not be possible here, since the `process` library would need to take a circular dependency on `builtin`. Or maybe I could simply have the `process` _tests_ dependent on `builtin`...?

Anyway, let me know if I'm missing something. I assume there's a better way that doesn't involve using a real filesystem, but I'm not sure what that is.